samples/audio-video-player/mp4_pull_demuxer.js

```diff
@@ -0,0 +1,254 @@
+const ENABLE_DEBUG_LOGGING = false;
+
+function debugLog(msg) {
+  if (!ENABLE_DEBUG_LOGGING) {
+    return;
+  }
+  console.debug(msg);
+}
+
+// Wrapper around MP4Box.js that shims pull-based demuxing on top their
+// push-based API.
+export class MP4PullDemuxer {
+  constructor(uri) {
+    this.source = new MP4Source(uri);
+    this.readySamples = [];
+    this._pending_read_resolver = null;
+  }
+
+  getAvcDescription(avccBox) {
```

Comment:
Nit: See my webgpu branch for a simpler approach (https://github.com/sandersdan/webcodecs/blob/webgpu/samples/mp4-decode/mp4_demuxer.js).

Reply:
Done!